packages/backend/src/tokens/authStatus.ts

```diff
@@ -14,7 +27,13 @@ export const AuthStatus = {
 
 export type AuthStatus = (typeof AuthStatus)[keyof typeof AuthStatus];
 
-export type SignedInState = {
+type ToAuth<T extends TokenType, Authenticated extends boolean> = T extends SessionTokenType
+  ? () => Authenticated extends true ? SignedInAuthObject : SignedOutAuthObject
+  : () => Authenticated extends true
+      ? AuthenticatedMachineObject<Exclude<T, SessionTokenType>>
+      : UnauthenticatedMachineObject<Exclude<T, SessionTokenType>>;
+
+export type AuthenticatedState<T extends TokenType = SessionTokenType> = {
```

Comment:
Deprecating `SignedInState`/`SignedOutState` in favor of`AuthenticatedState`/`UnauthenticatedState` to better reflect that authentication can happen via different token types (not just sessions).

These states are generic types defaulting to `SessionTokenType` for backwards compatibility. The generic type parameter flows through to `toAuth()` to return the correct auth object type (`SignedInAuthObject`/`SignedOutAuthObject` for sessions, `AuthenticatedMachineObject`/`UnauthenticatedMachineObject`for machine tokens).